### ISSUE TITLE
Cmake cross compiler ARM (C++) in WSL ubuntu 20.04

### ISSUE THREAD

**Issue by scizors** [opencv/opencv#19920]
Hi,
 I am trying to do cross compile for ARM for C++
 platform: WSL (ubuntu 20.04)
 
 I using latest opencv and opencv contrib from git
 
 I also followed this document but failed
 https://docs.opencv.org/master/d0/d76/tutorial_arm_crosscompile_with_cmake.html
  
 Issue in Cmake command
 `xxx@LD-MZ-NB-06:~/opencv_latest/build$ cmake -D CMAKE_BUILD_TYPE=Release -DCMAKE_EXTRA_MODULES_PATH=../opencv_contrib/modules/ -D CMAKE_INSTALL_PREFIX=/usr/local ../opencv -D BUILD_opencv_java=OFF -D BUILD_opencv_python2=OFF -D BUILD_opencv_python3=OFF -DSOFTFP=ON  -D CMAKE_TOOLCHAIN_FILE=../opencv/platforms/linux/arm-gnueabi.toolchain.cmake
-- Detected processor: arm
-- Could NOT find PythonInterp (missing: PYTHON_EXECUTABLE) (Required is at least version "2.7")
-- Looking for ccache - not found
-- NEON is not supported by C++ compiler
-- FP16 is not supported by C++ compiler
-- OpenCV disables pkg-config to avoid using of host libraries. Consider using PKG_CONFIG_LIBDIR to specify target SYSROOT`

**Comment by scizors** [comment]
is thi correct?

cmake -DSOFTFP=ON  -D CMAKE_TOOLCHAIN_FILE=../opencv/platforms/linux/arm-gnueabi.toolchain.cmake

**Comment by scizors** [comment]
```
[ 31%] Linking CXX shared library ../../lib/libopencv_imgcodecs.so
[ 32%] Built target opencv_imgcodecs
[ 32%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_gstreamer.cpp.o
In file included from /usr/lib/x86_64-linux-gnu/glib-2.0/include/glibconfig.h:9,
                 from /usr/include/glib-2.0/glib/gtypes.h:32,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/gstreamer-1.0/gst/gst.h:27,
                 from /home/sra/opencv_latest/opencv/modules/videoio/src/cap_gstreamer.cpp:59:
/usr/include/glib-2.0/glib/gtypes.h: In function ‘gboolean _GLIB_CHECKED_ADD_U64(guint64*, guint64, guint64)’:
/usr/include/glib-2.0/glib/gtypes.h:463:3: error: static assertion failed: Expression evaluates to false
  463 |   G_STATIC_ASSERT(sizeof (unsigned long long) == sizeof (guint64));
      |   ^~~~~~~~~~~~~~~
/home/sra/opencv_latest/opencv/modules/videoio/src/cap_gstreamer.cpp: In member function ‘bool cv::GStreamerCapture::isPipelinePlaying()’:
/home/sra/opencv_latest/opencv/modules/videoio/src/cap_gstreamer.cpp:601:29: warning: integer overflow in expression of type ‘long int’ results in ‘705032704’ [-Woverflow]
  601 |     GstClockTime timeout = 5*GST_SECOND;
      |                             ^
make[2]: *** [modules/videoio/CMakeFiles/opencv_videoio.dir/build.make:193: modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_gstreamer.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2776: modules/videoio/CMakeFiles/opencv_videoio.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

**Comment by scizors** [comment]
cmake -DSOFTFP=ON -D OPENCV_GENERATE_PKGCONFIG=YES -DCMAKE_TOOLCHAIN_FILE=./opencv/platforms/linux/arm-gnueabi.toolchain.cmake ../opencv
